### PR TITLE
Created Constructor to allow TcpConnect timeout to be set

### PR DESCRIPTION
**Pull request by glademiller**

### CODE REVIEW

File: src/actors/resolver.rs [hunk 1]
```diff
@@ -74,14 +79,43 @@ impl Message for Resolve {
 pub struct Connect {
     name: String,
     port: Option<u16>,
+    timeout: Option<Duration>,
```
Comment (fafhrd91) [inline src/actors/resolver.rs]: let's just  use `Duration`

File: src/actors/resolver.rs [hunk 2]
```diff
@@ -74,14 +79,43 @@ impl Message for Resolve {
...
     pub fn host_and_port<T: AsRef<str>>(host: T, port: u16) -> Connect {
-        Connect{name: host.as_ref().to_owned(), port: Some(port)}
+        Connect {
+            name: host.as_ref().to_owned(),
+            port: Some(port),
+            timeout: None,
+        }
+    }
+
+    pub fn host_and_timeout<T: AsRef<str>>(host: T, timeout: Duration) -> Connect {
```
Comment (fafhrd91) [inline src/actors/resolver.rs]: let's use builder pattern here. 

```rust
pub fn timeout(mut self, timeout: Duration) -> Connect {
    self.timeout = timeout;
    self
}
```

File: src/actors/resolver.rs [hunk 3]
```diff
@@ -74,14 +79,43 @@ impl Message for Resolve {
...
+        Connect {
+            name: host.as_ref().to_owned(),
+            port: None,
+            timeout: Some(timeout),
+        }
+    }
+
+    pub fn host_and_port_and_timeout<T: AsRef<str>>(
```
Comment (fafhrd91) [inline src/actors/resolver.rs]: we dont need this method with builder

File: src/actors/resolver.rs [hunk 4]
```diff
@@ -244,9 +287,11 @@ impl ActorFuture for Resolver {
     type Error = ConnectorError;
     type Actor = Connector;
 
-    fn poll(&mut self, _: &mut Connector, _: &mut Context<Connector>)
-            -> Poll<Self::Item, Self::Error>
-    {
+    fn poll(
```
Comment (fafhrd91) [inline src/actors/resolver.rs]: i dont like how this formatting looks

Comment (glademiller) [reply src/actors/resolver.rs]: Oops yep I'll turn off autoformatting

**Comment by fafhrd91** [comment]
Could you also update Changes.md

**Pull request merged**